src/encoded/static/components/image.js

```diff
@@ -5,177 +5,149 @@ import * as globals from './globals';
-            attachmentHref = url.resolve(context['@id'], attachment.href);
-            const attachmentType = attachment.type.split('/', 1)[0];
-            if (attachmentType === 'image' && attachment.type !== 'image/tiff') {
-                const imgClass = className ? `${className}-img` : '';
-                src = attachmentHref;
-                height = attachment.height || 100;
-                width = attachment.width || 100;
-                alt = 'Attachment from submitter';
-                if (!showLink) {
-                    // Just display the attachment image without any lightbox.
-                    return <img className={imgClass} src={src} height={height} width={width} alt={alt} />;
-                }
+        setLightboxVisible(true);
+    };
+
+    // Close the lightbox and image.
+    const clearLightbox = () => {
+        setLightboxVisible(false);
+    };
+
+    let attachmentHref;
+    let src;
+    let alt;
+    let height = '100';
+    let width = '100';
+    if (attachment && attachment.href && attachment.type) {
+        attachmentHref = url.resolve(context['@id'], attachment.href);
+        const attachmentType = attachment.type.split('/', 1)[0];
+        if (attachmentType === 'image' && attachment.type !== 'image/tiff') {
+            const imgClass = className ? `${className}-img` : '';
+            src = attachmentHref;
+            height = attachment.height || 100;
+            width = attachment.width || 100;
+            alt = 'Attachment from submitter';
+            if (!showLink) {
+                // Just display the attachment image without any lightbox.
+                return <img className={imgClass} src={src} height={height} width={width} alt={alt} />;
+            }
 
-                // Display the attachment image in a light box.
-                return (
-                    <div>
-                        <div className="attachment">
-                            <a className="attachment__button" data-bypass="true" href={attachmentHref} onClick={this.lightboxClickImage} title="View attachment in this window">
-                                <div className="attachment__hover" />
-                                <img className={imgClass} src={src} height={height} width={width} alt={alt} />
-                            </a>
-                        </div>
-                        <Lightbox lightboxVisible={this.state.lightboxVisible} lightboxImg={attachmentHref} clearLightbox={this.clearLightbox} />
-                    </div>
-                );
-            } else if (attachment.type === 'application/pdf') {
-                // Attachment is a PDF. Show the PDF icon, and clicks in it show the PDF in a new
-                // window using the browser's PDF viewer.
-                return (
+            // Display the attachment image in a light box.
+            return (
+                <div>
                     <div className="attachment">
```

Comment:
I have seen people use something like "js-attachment" for css classes that are used for javascript markers rather than styling. Is that a good approach to adopt.

Reply:
Haven’t thought about that specifically. `.attachment` has actual styles associated with it. Is it used as a Javascript maker? I don’t see where that happens, but I haven‘t looked at this code in a long time.